dev-util/hip/hip-5.0.2-r3.ebuild

```diff
@@ -24,6 +24,7 @@ SLOT="0/$(ver_cut 1-2)"
 IUSE="debug profile"
 
 DEPEND="
+	<sys-devel/gcc-12:*
 	dev-util/rocminfo:${SLOT}
 	=sys-devel/llvm-roc-${PV}*[runtime]
 	dev-libs/rocm-comgr:${SLOT}
```

Comment:
1. This should be BDEPEND
2. You must check in pkg_pretend (and pkg_setup) whether the _active_ compiler matches this, otherwise someone with e.g. GCC 11 installed but GCC 12 selected will end up hitting a build failure. (Lots of examples of this in tree, go grep)

Reply:
(Also, obviously, this is kind of a grim workaround and I really hope this isn't going to last long.)